src/base/bittorrent/bencoderesumedatastorage.cpp

```diff
@@ -151,25 +150,36 @@ std::optional<BitTorrent::LoadTorrentParams> BitTorrent::BencodeResumeDataStorag
     const QString fastresumePath = m_resumeDataDir.absoluteFilePath(QString::fromLatin1("%1.fastresume").arg(idString));
     const QString torrentFilePath = m_resumeDataDir.absoluteFilePath(QString::fromLatin1("%1.torrent").arg(idString));
 
-    QFile file {fastresumePath};
-    if (!file.open(QIODevice::ReadOnly))
+    QFile resumeDataFile {fastresumePath};
+    if (!resumeDataFile.open(QIODevice::ReadOnly))
     {
-        LogMsg(tr("Cannot read file %1: %2").arg(fastresumePath, file.errorString()), Log::WARNING);
+        LogMsg(tr("Cannot read file %1: %2").arg(fastresumePath, resumeDataFile.errorString()), Log::WARNING);
         return std::nullopt;
     }
 
-    const QByteArray data = file.readAll();
-    const TorrentInfo metadata = TorrentInfo::loadFromFile(torrentFilePath);
+    QFile metadataFile {torrentFilePath};
+    if (metadataFile.exists() && !metadataFile.open(QIODevice::ReadOnly))
+    {
+        LogMsg(tr("Cannot read file %1: %2").arg(torrentFilePath, metadataFile.errorString()), Log::WARNING);
+        return std::nullopt;
+    }
+
+    const QByteArray data = resumeDataFile.readAll();
+    const QByteArray metadata = (metadataFile.isOpen() ? metadataFile.readAll() : "");
```

Comment:
maybe use `{}`

Reply:
It won't compile.

Reply:
> Maybe we will not have to resort to this workaround, if [arvidn/libtorrent#6283](https://github.com/arvidn/libtorrent/issues/6283) is resolved in a positive way for us.

But I don't think we should wait for it. This may take an indefinite period of time, and besides, we cannot predict its outcome.